metrics/api/README.md

```diff
@@ -0,0 +1,206 @@
+components. 
+It provides a `RegistryFactory` interface which is API-compatible with and expands on the earlier 
+`RegistryFactory` 
+class in `helidon-metrics`. (The `helidon-metrics` `RegistryFactory` static methods now delegate 
+to the `helidon-metrics-api` `RegistryFactory` to allow components to migrate to the new 
+approach over time.)
+
+_Metrics-capable_ Helidon components are ones which should create and update metrics when a 
+metrics 
+implementation is on the runtime path and the user configuration so indicates. These components 
+need to 
+compile and run even if a developer:
+*  omits the metrics implementation altogether, so the application never supports 
+   full-featured metrics, or
+* includes the metrics implementation but uses configuration to disable 
+metrics for individual components or for the app as a whole.
+
+This `helidon-metrics-api` module allows the code in such components to use metrics classes and 
+interfaces 
+ regardless of whether metrics is or is not present on the path at build-time or runtime, and 
+regardless of whether metrics is disabled for a particular run of the application. If metrics is 
+present and enabled at runtime, then Helidon provides the component full-featured metrics; 
+otherwise it furnishes a "no-op" implementation.
+
+Key elements in this module for metrics-capable components:
+* `RegistryFactory` interface
+      
+   Components use this interface instead of the 
+  earlier `RegistryFactory` 
+  class from `helidon-metrics` which components have used 
+  previously.
+* `ComponentMetricsSettings` interface and its `Builder` interface
+  
+   Attributes of metrics for 
+  metrics-capable components (currently, 
+  simply whether metrics are to be enabled for the component or not). 
+
+## Converting a metrics-dependent component to a metrics-enabled one
+1. Change `pom.xml` to depend on `helidon-metrics-api` instead of `helidon-metrics`.
+2. Either:
+   1. Change the service to extend `HelidonRestServiceSuport` if it does not already, or
+   2. Update the service and the service's builder to handle a `ComponentMetricsSettings` object 
+   and a `metrics` config section within the component's config. (see below)
+3. Change invocations of static factory methods on the `helidon-metrics` `RegistryFactory` class to 
+   invoke `io.
+   helidon.metrics.api.RegistryFactory.instance(ComponentMetricsSettings)` instead.
+   
+   This method returns a `RegistryFactory` which automatically accounts for:
+   * whether the metrics implementation is available at runtime,
+   * whether the overall metrics configuration has enabled or disabled metrics in general, and
+   * whether the component's metrics settings indicate to use the full-featured metrics
+  implementation (assuming it is available).
+
+Aside from that, leave the metrics-related code in metrics-capable components as-is.
+
+For example, suppose the `helidon-xxx` component creates or updates some metrics. You have two 
```

Comment:
If this is for application developers then maybe `yourapp-xxx` instead of `helidon-xxx`?

Reply:
Gone do to other edits.